### ISSUE TITLE
Battle list is laggy with no filters and lots of battles

### ISSUE THREAD

**Issue by Jazcash** [beyond-all-reason/bar-lobby#192]
![image](https://user-images.githubusercontent.com/1434248/230676615-e1bfe494-5492-421f-887b-9826e8ab866a.png)

Scrolling up and down this isn't a fun time (~250 battles atm). We should either paginate it or use https://primevue.org/virtualscroller

**Comment by dusekj4** [comment]
will look into it

**Comment by Jazcash** [comment]
Looked into this to give you some notes, found that the `DataTable` component has aa `virtualScrollerOptions` prop which _would_ solve this issue, if it wasn't for the fact we're not actually using the datatable's built-in scroll functionality. if you add `scrollable :scrollHeight="'500px'"` and remove the `scroll-container` parent div, you'll see the datatable's scroll. Only problem with that is it doesn't grow to fill the remaining space and seems to require a fixed height value. Also the table's headers overlap with the rows because of the semi-transparent styling. It also seems to scroll back to the top whenever the data is updated

**Comment by Jazcash** [comment]
ack sorry I just ended up doing this 😬

**Comment by Jazcash** [comment]
Gonna reopen this because I've just noticed it lagging again, even though it was much smoother the other day, even though I tripled the data when testing it. I'm also not too satisfied with this virtual scrolling part of DataTable in general - it jumps around quite a bit (probably when the data updates), sometimes scrolls back to the top, isn't the smoothest thing ever, the headings have to have solid colour background because the data is underneath...

[ice_video_20230423-234525.webm](https://user-images.githubusercontent.com/1434248/233870168-0554244a-e9a3-411a-9ee5-c937a689cf9a.webm)

Tempted to paginate for now. In the long-run, we should expect this list to (hopefully) be much smaller when (if) people use matchmaking instead. Private battles should also eventually be hidden when we have better ways for friends to join each other or send out invites.
